src/mcmini_private.cpp

```diff
@@ -34,6 +34,26 @@ sem_t mc_pthread_create_binary_sem;
 trid_t traceId      = 0;
 trid_t transitionId = 0;
 
+static char resultString[1000] = "***** Model checking completed! *****\n";
+static void addResult(const char *result) {
+  char stats[1000];
+  if (strstr(resultString, result) != NULL) {
+    result = "  (Other trace numbers exist: ...)\n";
+    if (strstr(resultString, result) == NULL) {
+      strncat(resultString, result, sizeof(resultString) - strlen(resultString));
+    }
+    return;
+  }
+  strncat(resultString, result, sizeof(resultString) - strlen(resultString));
+  snprintf(stats, 80, "  (Trace number: %lu)\n", traceId);
+  strncat(resultString, stats, sizeof(resultString) - strlen(resultString));
+}
+static void printResults() {
+  mcprintf(resultString);
```

Comment:
Every function in McMini could be labelled as `mc_*`.  I don't see how it makes the code clearer if every function has `mc_` as a prefix.  :-)

Usually, a prefix like `mc_` is added so that we don't contaminate the target application.  If McMini defines `foo()`, there is a danger that the target will also call `foo()`.  But if McMini defines `mc_foo()`, then there is no conflict with the target.  Here, we declare `printResults` and `addResult` as static (file-local), and so there is no possible conflict.

In general, one can declare a function as file-local (global static), library-local (`void __attribute__ ((visibility ("protected")))`), and non-local.  The non-local functions need a prefix like `mc_`.

But would you like to propose a name beside `printResults()` and `addResult()`?